### PR TITLE
Remove some Pytest specific libraries and use stdlib instead. Prevent name collisions in schema.

### LINKED ISSUE #25: Removal of 'py' dependencies
Due to using some of pytest's internals there is some stuff we use from pytest's `py` library.

`py` is not very nice. It avoids the stdlib (by not using `tempfile` for example) and has implemented a lot of its own methods. 

I would prefer to use stdlib methods wherever possible, to make our code more readable and maintainable. I am thinking specifically of the temporary directory for worklows. This is now implemented using `pytest`'s methods. This was done in the hope that pytest would handle cleanup automatically. Since we have realized cleanup explicitly, the use of pytest's internal methods is no longer necessary and we can use `shutil.copytree`, `shutil.rmtree` and `tempfile.mkdtemp` instead.

### PR DESCRIPTION
**Pull request by rhpvorderman**
All the tempdir functionality is now stdlib stuff.
Turns out that even if you prevent colliding directory names that pytest will still crash if test names are the same. So I threw out stuff to prevent directory collisions and implemented a simple schema check instead.

fixes #25 

### Checklist
- [x] Pull request details were added to HISTORY.rst

### CODE REVIEW

File: src/pytest_workflow/workflow.py [hunk 1]
```diff
@@ -23,13 +23,12 @@
...
 import subprocess  # nosec: security implications have been considered
 from pathlib import Path
-from typing import Union
 
 
 class Workflow(object):
 
     def __init__(self, command: str,
-                 cwd: Union[bytes, str] = None):
+                 cwd: Path = Path("")):
```
Comment (rhpvorderman) [inline src/pytest_workflow/workflow.py]: I am a bit uncertain whether this is a correct default.

File: src/pytest_workflow/plugin.py [hunk 2]
```diff
@@ -138,7 +131,9 @@ def run_workflow(self):
         else:
             # addfinalizer adds a function that is run when the node tests are
             # completed
-            self.addfinalizer(tempdir.remove)
+            def rm_tempdir():
+                shutil.rmtree(str(tempdir))
+            self.addfinalizer(rm_tempdir)
```
Comment (sndrtj) [inline src/pytest_workflow/plugin.py]: Perhaps in stead of defining a function inside of a function (hairy imho) you could opt to use functools partial functions. E.g.

```python
from functools import partial

finalizer_func = partial(shutil.rmtree, str(tempdir))
self.addfinalizer(finalizer_func)
```

File: src/pytest_workflow/schema.py [hunk 3]
```diff
@@ -46,6 +48,16 @@ def validate_schema(instance):
     jsonschema.validate(instance, JSON_SCHEMA)
 
     # Some extra tests here below that can not be captured in jsonschema
+
+    # Test if there are name collisions when whitespace is removed. This will
+    # cause errors in pytest (collectors not having unique names) so has to
+    # be avoided.
+    test_names = [replace_whitespace(test['name'], ' ') for test in instance]
```
[the remark this answers is not in the document]
Comment (rhpvorderman) [reply src/pytest_workflow/schema.py]: This is for purely cosmetic reasons in the resulting validation error.

[the remark this answers is not in the document]
Comment (rhpvorderman) [reply src/pytest_workflow/schema.py]: i.e "bla      bla" -> "bla bla"
instead of "blabla" or "bla_bla"

File: src/pytest_workflow/plugin.py (hunk 2 above)
Comment (rhpvorderman) [reply src/pytest_workflow/plugin.py]: :+1:
Great. I have seen this before and I believe this is the idiomatic way.

**Pull request merged**